py/picca/bal_tools.py

```diff
@@ -46,17 +44,22 @@ def read_bal(filename,survey):  ##Based on read_dla from picca/py/picca/io.py
 
     return bal_catalog
 
-def add_bal_mask(bal_catalog, objectid, id_name='TARGETID'):
+def add_bal_mask(bal_catalog, objectid, survey):
     """Creates a list of wavelengths to be masked out by forest.mask
 
     Args:
         bal_catalog: str
             Catalog of BALs
         objectid: str
             Identifier of quasar
-        id_name: str
-            Column name of quasar identification ("THING_ID" in EBOSS)
+        survey: str
+            Survey catalog comes from. 
     """
+    if survey == 'eboss':
+        id_name = 'THING_ID'
+    else:
```

Comment:
the `else` should be `elif survey == 'desi'` and there should be an extra `else` with an error

Reply:
addressed in 06a81ce